projects/ngx-datatable/src/lib/components/body/body.component.ts

```diff
@@ -471,6 +487,85 @@ export class DataTableBodyComponent<TRow extends { treeStatus?: TreeStatus } = a
     }
   }
 
+  private updateColumnsByPin() {
+    this._columnsByPin = columnsByPinArr(this.columns);
+  }
+
+  private updateGroupStyles() {
+    this._groupStyles.left = this.calcStylesByGroup('left');
+    this._groupStyles.center = this.calcStylesByGroup('center');
+    this._groupStyles.right = this.calcStylesByGroup('right');
+    this.cd.markForCheck();
+  }
+
+  private calcStylesByGroup(group: 'left' | 'right' | 'center') {
+    const widths = this.columnGroupWidths;
+    const offsetX = this.offsetX;
+
+    if (group === 'left') {
+      return {
+        width: `${widths[group]}px`,
+        ...translateXY(offsetX, 0)
+      };
+    } else if (group === 'right') {
+      const bodyWidth = this.innerWidth;
+      const totalDiff = widths.total - bodyWidth;
+      const offsetDiff = totalDiff - offsetX;
+      const offset =
+        (offsetDiff + (this.verticalScrollVisible ? this.scrollbarHelper.getWidth() : 0)) * -1;
+      return {
+        width: `${widths[group]}px`,
+        ...translateXY(offset, 0)
+      };
+    }
+
+    return {
+      width: `${widths[group]}px`
+    };
+  }
```

Comment:
@chintankavathia Makes sense, but the main part of this still stands, which is to avoid doing calculations in every row and doing them shared in the body component instead.